// primeng/app/app.component.ts
import {Component, OnInit} from '@angular/core';
import {Http} from '@angular/http';
import {URLSearchParams, QueryEncoder} from '@angular/http';
import 'rxjs/add/operator/toPromise';
import {SelectItem} from 'primeng/primeng';

@Component({
selector: 'my-app',
templateUrl: './app/app.component.html',
})
export class AppComponent implements OnInit{

    monthTopic : string = "1";
    yearTopic : string = "2016";
    topic : string;

    monthArea : string = "1";
    yearArea : string = "2016";
    area : string;

    yearPerMonth : string = "2016";

    monthPerDay : string = "1";
    yearPerDay : string = "2016";

    dataTopic:any;
    dataArea:any;
    dataPerMonth:any;
    dataPerDay:any;

    topicsMonths: SelectItem[];
    topicsYears : SelectItem[];

    areasMonths : SelectItem[];
    areasYears : SelectItem[];

    perMonthYears : SelectItem[];

    perDayMonths : SelectItem[];
    perDayYears : SelectItem[];

    constructor(private http: Http){
        this.dataTopic = {
            'labels': ['1', '2', '3', '4', '5', '6', '7', '8', '9', '10', '11', '12', '13', '14', '15', '16', '17', '18', '19', '20', '21', '22', '23', '24', '25', '26', '27', '28', '29', '30', '31'],
            'datasets': [
                {
                    'label': 'First Dataset',
                    'data': [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]
                }
            ]
        }

        this.dataArea = {
            'labels': ['1', '2', '3', '4', '5', '6', '7', '8', '9', '10', '11', '12', '13', '14', '15', '16', '17', '18', '19', '20', '21', '22', '23', '24', '25', '26', '27', '28', '29', '30', '31'],
            'datasets': [
                {
                    'label': 'First Dataset',
                    'data': [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]
                }
            ]
        }

        this.dataPerMonth = {
            'labels': ['1', '2', '3', '4', '5', '6', '7', '8', '9', '10', '11', '12', '13', '14', '15', '16', '17', '18', '19', '20', '21', '22', '23', '24', '25', '26', '27', '28', '29', '30', '31'],
            'datasets': [
                {
                    'label': 'First Dataset',
                    'data': [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]
                }
            ]
        }

        this.dataPerDay = {
            'labels': ['1', '2', '3', '4', '5', '6', '7', '8', '9', '10', '11', '12', '13', '14', '15', '16', '17', '18', '19', '20', '21', '22', '23', '24', '25', '26', '27', '28', '29', '30', '31'],
            'datasets': [
                {
                    'label': 'First Dataset',
                    'data': [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]
                }
            ]
        }

        this.topicsMonths = [];
        this.topicsMonths.push({label:"January",value:1});
        this.topicsMonths.push({label:"February",value:2});
        this.topicsMonths.push({label:"March",value:3});
        this.topicsMonths.push({label:"April",value:4});
        this.topicsMonths.push({label:"May",value:5});
        this.topicsMonths.push({label:"June",value:6});
        this.topicsMonths.push({label:"July",value:7});
        this.topicsMonths.push({label:"August",value:8});
        this.topicsMonths.push({label:"September",value:9});
        this.topicsMonths.push({label:"October",value:10});
        this.topicsMonths.push({label:"November",value:11});
        this.topicsMonths.push({label:"December",value:12});

        this.topicsYears = [];
        this.topicsYears.push({label:"2016",value:2016});
        this.topicsYears.push({label:"2015",value:2015});
        this.topicsYears.push({label:"2014",value:2014});
        this.topicsYears.push({label:"2013",value:2013});
        this.topicsYears.push({label:"2012",value:2012});
        this.topicsYears.push({label:"2011",value:2011});
        this.topicsYears.push({label:"2010",value:2010});
        this.topicsYears.push({label:"2009",value:2009});
        this.topicsYears.push({label:"2008",value:2008});
        this.topicsYears.push({label:"2007",value:2007});

        this.areasMonths = [];
        this.areasMonths.push({label:"January",value:1});
        this.areasMonths.push({label:"February",value:2});
        this.areasMonths.push({label:"March",value:3});
        this.areasMonths.push({label:"April",value:4});
        this.areasMonths.push({label:"May",value:5});
        this.areasMonths.push({label:"June",value:6});
        this.areasMonths.push({label:"July",value:7});
        this.areasMonths.push({label:"August",value:8});
        this.areasMonths.push({label:"September",value:9});
        this.areasMonths.push({label:"October",value:10});
        this.areasMonths.push({label:"November",value:11});
        this.areasMonths.push({label:"December",value:12});

        this.areasYears = [];
        this.areasYears.push({label:"2016",value:2016});
        this.areasYears.push({label:"2015",value:2015});
        this.areasYears.push({label:"2014",value:2014});
        this.areasYears.push({label:"2013",value:2013});
        this.areasYears.push({label:"2012",value:2012});
        this.areasYears.push({label:"2011",value:2011});
        this.areasYears.push({label:"2010",value:2010});
        this.areasYears.push({label:"2009",value:2009});
        this.areasYears.push({label:"2008",value:2008});
        this.areasYears.push({label:"2007",value:2007});

        this.perMonthYears = [];
        this.perMonthYears.push({label:"2016",value:2016});
        this.perMonthYears.push({label:"2015",value:2015});
        this.perMonthYears.push({label:"2014",value:2014});
        this.perMonthYears.push({label:"2013",value:2013});
        this.perMonthYears.push({label:"2012",value:2012});
        this.perMonthYears.push({label:"2011",value:2011});
        this.perMonthYears.push({label:"2010",value:2010});
        this.perMonthYears.push({label:"2009",value:2009});
        this.perMonthYears.push({label:"2008",value:2008});
        this.perMonthYears.push({label:"2007",value:2007});

        this.perDayMonths = [];
        this.perDayMonths.push({label:"January",value:1});
        this.perDayMonths.push({label:"February",value:2});
        this.perDayMonths.push({label:"March",value:3});
        this.perDayMonths.push({label:"April",value:4});
        this.perDayMonths.push({label:"May",value:5});
        this.perDayMonths.push({label:"June",value:6});
        this.perDayMonths.push({label:"July",value:7});
        this.perDayMonths.push({label:"August",value:8});
        this.perDayMonths.push({label:"September",value:9});
        this.perDayMonths.push({label:"October",value:10});
        this.perDayMonths.push({label:"November",value:11});
        this.perDayMonths.push({label:"December",value:12});

        this.perDayYears = [];
        this.perDayYears.push({label:"2016",value:2016});
        this.perDayYears.push({label:"2015",value:2015});
        this.perDayYears.push({label:"2014",value:2014});
        this.perDayYears.push({label:"2013",value:2013});
        this.perDayYears.push({label:"2012",value:2012});
        this.perDayYears.push({label:"2011",value:2011});
        this.perDayYears.push({label:"2010",value:2010});
        this.perDayYears.push({label:"2009",value:2009});
        this.perDayYears.push({label:"2008",value:2008});
        this.perDayYears.push({label:"2007",value:2007});
    }

    ngOnInit(){
        
    }

    onclicktopic() {
        if(this.topic != "undefined" && this.topic != "" && this.topic != null){
            let params = new URLSearchParams();
            params.set("month", this.monthTopic);
            params.set("year", this.yearTopic);
            params.set("topics", this.topic);
            
            this.http.get("http://localhost:8116/mostMentionedTopics", { search : params}).
                subscribe( data => {this.dataTopic = JSON.parse(data['_body'])} , err => console.error(err), () => console.log("done"));
        }else{
            alert("Please Fill In The Topic Box");
        }
    }

    onclickarea(){
        if(this.area != "undefined" && this.area != "" && this.area != null){
            let params = new URLSearchParams();
            params.set("month", this.monthArea);
            params.set("year", this.yearArea);
            params.set("areas", this.area);

            this.http.get("http://localhost:8116/mostMentionedAreas", { search : params}).
                subscribe( data => {this.dataArea = JSON.parse(data['_body'])} , err => console.error(err), () => console.log("done"));
        }else{
            alert("Please Fill In The Area Box");
        }
    }

    onclickpermonth(){
        let params = new URLSearchParams();
        params.set("year", this.yearPerMonth);

        this.http.get("http://localhost:8116/totalCasePerMonth", { search : params}).
            subscribe( data => {this.dataPerMonth = JSON.parse(data['_body'])} , err => console.error(err), () => console.log("done"));
    }

    onclickperday(){
        let params = new URLSearchParams();
        params.set("month", this.monthPerDay);
        params.set("year", this.yearPerDay);

        this.http.get("http://localhost:8116/totalCasePerDay", { search : params}).
            subscribe( data => {this.dataPerDay = JSON.parse(data['_body'])} , err => console.error(err), () => console.log("done"));
    }
    
}
