// primeng/app/chart-perday.component.js
"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
var core_1 = require('@angular/core');
var http_1 = require('@angular/http');
require('rxjs/add/operator/toPromise');
var ChartPerDayComponent = (function () {
    function ChartPerDayComponent(http) {
        /*
        this.data = {
            'labels': ['1', '2', '3', '4', '5', '6', '7', '8', '9', '10', '11', '12', '13', '14', '15', '16', '17', '18', '19', '20', '21', '22', '23', '24', '25', '26', '27', '28', '29', '30', '31'],
            'datasets': [
                {
                    'label': 'First Dataset',
                    'data': [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]
                }
            ]
        }
        */
        this.http = http;
        //this.http.get("http://localhost:8116/totalCasePerDay").
        //subscribe( data => {this.data = JSON.parse(data['_body'])} , err => console.error(err), () => console.log("done"));
    }
    ChartPerDayComponent.prototype.ngOnInit = function () {
    };
    __decorate([
        core_1.Input(), 
        __metadata('design:type', Object)
    ], ChartPerDayComponent.prototype, "data", void 0);
    ChartPerDayComponent = __decorate([
        core_1.Component({
            selector: 'chartPerDay',
            template: '<p-chart type="line" [data]="data"></p-chart>',
        }), 
        __metadata('design:paramtypes', [http_1.Http])
    ], ChartPerDayComponent);
    return ChartPerDayComponent;
}());
exports.ChartPerDayComponent = ChartPerDayComponent;
